wikipedia_connector/wiki.py

```diff
@@ -99,48 +82,77 @@ def get_clean_text(self, text):
+                    if(abc2.get_wiki_page(str(topics.columns.values[j]))[1] == '1'):
+                        disambiguation = disambiguation + 1
+                    list2.append(str(topics[i]))
+            #lst2 = ['x'] * len(input_list)
+            df1 = pd.DataFrame(list(zip(input_list, list2)), columns =['text', 'title'])
+            print('There were ' + str(disambiguation) + ' ambigious values in the input list')
+            output_summaries_file = "wiki_output.csv"
+            df1.to_csv(file_dir+"/{}".format(output_summaries_file), index=False)
         else:
-            for j in range(len(topics.columns.values)):
-                abc2 = WikiPage(str(topics.columns.values[j]))
-                abc22 = abc2.get_wiki_page(str(topics.columns.values[j]))[0]
-                input_list.append(abc22)
-                if(abc2.get_wiki_page(str(topics.columns.values[j]))[1] == '1'):
+            input_list = []
+            list2 = []
+            disambiguation = 0
+            for i in range(len(topics)):
+                abc1 = WikiPage(str(topics[i]))
+                abc11 = abc1.get_wiki_page(str(topics[i]))[0]
+                input_list.append(abc11)
+                if(abc1.get_wiki_page(str(topics[i]))[1] == '1'):
                     disambiguation = disambiguation + 1
                 list2.append(str(topics[i]))
-        #lst2 = ['x'] * len(input_list)
-        df1 = pd.DataFrame(list(zip(input_list, list2)), columns =['text', 'title'])
-        print('There were ' + str(disambiguation) + ' ambigious values in the input list')
-        output_summaries_file = "wiki_output.csv"
-        df1.to_csv(cnvrg_workdir+"/{}".format(output_summaries_file), index=False)
-    else:
-        input_list = []
-        list2 = []
-        disambiguation = 0
-        for i in range(len(topics)):
-            abc1 = WikiPage(str(topics[i]))
-            abc11 = abc1.get_wiki_page(str(topics[i]))[0]
-            input_list.append(abc11)
-            if(abc1.get_wiki_page(str(topics[i]))[1] == '1'):
-                disambiguation = disambiguation + 1
-            list2.append(str(topics[i]))
-        #lst2 = ['x'] * len(input_list)
-        df1 = pd.DataFrame(list(zip(input_list, list2)), columns =['text', 'title'])
-        print('There were ' + str(disambiguation) + ' ambigious values in the input list')
-        output_summaries_file = "wiki_output.csv"
-        df1.to_csv(cnvrg_workdir+"/{}".format(output_summaries_file), index=False)
+            #lst2 = ['x'] * len(input_list)
+            df1 = pd.DataFrame(list(zip(input_list, list2)), columns =['text', 'title'])
+            print('There were ' + str(disambiguation) + ' ambigious values in the input list')
+            output_summaries_file = "wiki_output.csv"
+            df1.to_csv(file_dir+"/{}".format(output_summaries_file), index=False)
+
+    except (wikipedia.exceptions.PageError, wikipedia.exceptions.WikipediaException,urllib.error.HTTPError):
```

Comment:
recommend adding a negative testcase to catch this exception

Reply:
@orionsBeltWest Added the negative testcase and refactored more code